kong/db/dao/plugins/external.lua

```diff
@@ -0,0 +1,642 @@
+local cjson = require "cjson.safe"
+local ngx_ssl = require "ngx.ssl"
+local pl_path = require "pl.path"
+local raw_log = require "ngx.errlog".raw_log
+
+local rpc = require "kong.db.dao.plugins.mp_rpc"
+
+local ngx = ngx
+local kong = kong
+local unpack = unpack
+local ngx_INFO = ngx.INFO
+local ngx_timer_at = ngx.timer.at
+local cjson_encode = cjson.encode
+local cjson_decode = cjson.decode
+
+--- module table
+local external_plugins = {}
+
+
+local _servers
+local _plugin_infos
+
+--- keep request data a bit longer, into the log timer
+local save_for_later = {}
+
+--- handle notifications from pluginservers
+local rpc_notifications = {}
+
+--- currently running plugin instances
+local running_instances = {}
+
+
+--[[
+
+Configuration
+
+The external_plugins_config YAML file defines a list of plugin servers.  Each one
+can have the following fields:
+
+name: (required) a unique string.  Shouldn't collide with Lua packages available as `kong.plugins.<name>`
+socket: (required) path of a unix domain socket to use for RPC.
+exec: (optional) executable file of the server.  If omitted, the server process won't be managed by Kong.
```

Comment:
What is the difference between `exec` and `start_command`?

Reply:
scar tissue from the yaml->config change